crates/solver-core/src/lib.rs

```diff
@@ -669,6 +669,50 @@ impl SolverEngine {
 			}
 		};
 
+		// Store completion status in storage
+		self.storage
+			.store(
+				"completed_orders",
+				&order_id,
+				&std::time::SystemTime::now()
+					.duration_since(std::time::UNIX_EPOCH)
+					.unwrap()
+					.as_secs(),
+			)
+			.await
+			.map_err(|e| SolverError::Service(e.to_string()))?;
```

Comment:
Not sure if we need this, what other tables exist already? E.g. "orders"? Then we can just retrieve by status or something?